### PR TITLE
[4/N] make quant config first-class citizen

### PR DESCRIPTION
**Pull request by youkaichao**
continue of https://github.com/vllm-project/vllm/pull/9958 , clean up the config structure.

### CODE REVIEW

File: vllm/config.py [hunk 1]
```diff
@@ -1958,6 +1962,35 @@ class VllmConfig:
     decoding_config: Optional[DecodingConfig] = None
     observability_config: Optional[ObservabilityConfig] = None
     prompt_adapter_config: Optional[PromptAdapterConfig] = None
+    quant_config: Optional[QuantizationConfig] = field(init=False)
```
Comment (mgoin) [inline vllm/config.py:1965]: Why not default to None like the other configs?

**Review by mgoin — APPROVED** [review]
LGTM, thanks for this consolidation

File: vllm/config.py (hunk 1 above)
Comment (youkaichao) [reply vllm/config.py]: good point, changed in [a4995fa](https://github.com/vllm-project/vllm/pull/9978/commits/a4995fa3ab7670d9032c373f53ed421922b4830e)

**Comment by kylesayrs** [comment]
```
else:
    QuantizationConfig = None

**Comment by kylesayrs** [comment]
@youkaichao Why were these lines added?

```python3
else:
    QuantizationConfig = None
```

**Pull request merged**